src/ui/AdvancedSearch/DateInput/DateInput.ts

```diff
@@ -35,16 +36,36 @@ export class DateInput implements IAdvancedSearchInput {
   }
 
   public updateQuery(queryBuilder: QueryBuilder) {
-    let value = this.getValue();
-    if (value) {
-      queryBuilder.advancedExpression.add(this.getValue());
+    try {
+      let value = this.getValue();
+      if (value) {
+        queryBuilder.advancedExpression.add(this.getValue());
```

Comment:
Do we need to call `getValue` again here when update the advanced expression? Also, the `DateInput` class on its own makes no sense. The `getValue` method returns an empty string which would make the `updateQuery` method dead code. I think it should be an abstract class and the `getValue` method should be abstract.